pkg/appgw/backendhttpsettings.go

```diff
@@ -158,38 +160,44 @@ func (c *appGwConfigBuilder) BackendHTTPSettingsCollection(ingressList []*v1beta
 			uniquePair = k
 		}
 
-		c.serviceBackendPairMap[backendID] = uniquePair
+		finalServiceBackendPairMap[backendID] = uniquePair
 		httpSettings := c.generateHTTPSettings(backendID, uniquePair.BackendPort)
 		httpSettingsCollection[*httpSettings.Name] = httpSettings
-		c.backendHTTPSettingsMap[backendID] = &httpSettings
+		backendHTTPSettingsMap[backendID] = &httpSettings
 	}
 
 	backends := make([]network.ApplicationGatewayBackendHTTPSettings, 0, len(httpSettingsCollection))
 	for _, backend := range httpSettingsCollection {
 		backends = append(backends, backend)
 	}
 
-	c.appGwConfig.BackendHTTPSettingsCollection = &backends
+	return &backends, backendHTTPSettingsMap, finalServiceBackendPairMap, nil
+}
 
-	return nil
+func (c *appGwConfigBuilder) BackendHTTPSettingsCollection(ingressList []*v1beta1.Ingress) error {
+	backends, _, _, err := c.getBackendsAndSettingsMap(ingressList)
+	c.appGwConfig.BackendHTTPSettingsCollection = backends
+	return err
 }
 
 func (c *appGwConfigBuilder) generateHTTPSettings(backendID backendIdentifier, port int32) network.ApplicationGatewayBackendHTTPSettings {
-	probeName := c.probesMap[backendID].Name
-	probeID := c.appGwIdentifier.probeID(*probeName)
 	httpSettingsName := generateHTTPSettingsName(backendID.serviceFullName(), backendID.Backend.ServicePort.String(), port, backendID.Ingress.Name)
 	glog.Infof("Created a new HTTP setting w/ name: %s\n", httpSettingsName)
-
 	httpSettings := network.ApplicationGatewayBackendHTTPSettings{
 		Etag: to.StringPtr("*"),
 		Name: &httpSettingsName,
 		ApplicationGatewayBackendHTTPSettingsPropertiesFormat: &network.ApplicationGatewayBackendHTTPSettingsPropertiesFormat{
 			Protocol: network.HTTP,
 			Port:     &port,
-			Probe:    resourceRef(probeID),
 		},
 	}
 
+	if c.probesMap[backendID] != nil {
```

Comment:
This is a great improvement!
We should have a default probe -- but it is a great idea to catch this in cases where we have probes.